internals/secrethub/service_init_test.go

```diff
@@ -0,0 +1,244 @@
+			},
+			serviceService: fakeclient.ServiceService{
+				CreateFunc: func(path string, description string, credentialCreator credentials.Creator) (*api.Service, error) {
+					return &api.Service{
+						ServiceID: "testService",
+					}, nil
+				},
+				DeleteFunc: func(id string) (*api.RevokeRepoResponse, error) {
+					return &api.RevokeRepoResponse{}, nil
+				},
+			},
+			setFunc: func(path string, permission string, accountName string) (*api.AccessRule, error) {
+				return &api.AccessRule{}, testErr
+			},
+			err: testErr,
+		},
+		"fail permission and revoke": {
+			cmd: ServiceInitCommand{
+				repo:       api.RepoPath("test/repo"),
+				credential: keyCreator,
+				permission: "read",
+			},
+			serviceService: fakeclient.ServiceService{
+				CreateFunc: func(path string, description string, credentialCreator credentials.Creator) (*api.Service, error) {
+					return &api.Service{
+						ServiceID: "testService",
+					}, nil
+				},
+				DeleteFunc: func(id string) (*api.RevokeRepoResponse, error) {
+					return nil, fmt.Errorf("revoke has failed")
+				},
+			},
+			setFunc: func(path string, permission string, accountName string) (*api.AccessRule, error) {
+				return &api.AccessRule{}, testErr
+			},
+			err: fmt.Errorf("revoke has failed"),
+		},
+		"fail no key created": {
+			cmd: ServiceInitCommand{
+				repo: api.RepoPath("test/repo"),
+			},
+			serviceService: fakeclient.ServiceService{
+				CreateFunc: func(path string, description string, credentialCreator credentials.Creator) (*api.Service, error) {
+					return &api.Service{}, nil
+				},
+			},
+			err: fmt.Errorf("key has not yet been generated created. Use KeyCreator before calling Export()"),
+		},
+		"init fail file exists": {
+			cmd: ServiceInitCommand{
+				file: "test.txt",
+			},
+			err: ErrFileAlreadyExists,
+		},
+		"init fail flags conflict": {
+			cmd: ServiceInitCommand{
+				clip: true,
+				file: "test.txt",
+			},
+			err: ErrFlagsConflict("--clip and --file"),
+		},
+		"new client error": {
+			newClientErr: testErr,
+			err:          testErr,
+		},
+		"service init error": {
+			serviceService: fakeclient.ServiceService{
+				CreateFunc: func(path string, description string, credentialCreator credentials.Creator) (*api.Service, error) {
+					return nil, testErr
+				},
+			},
+			err: testErr,
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			// Setup
+			var perm *api.AccessRule
+			var fileOut []byte
+			testIO := fakeui.NewIO(t)
+			tc.cmd.io = testIO
+
+			tc.cmd.newClient = func() (secrethub.ClientInterface, error) {
+				return fakeclient.Client{
+					ServiceService: &tc.serviceService,
+					AccessRuleService: &fakeclient.AccessRuleService{
+						SetFunc: func(path string, permission string, accountName string) (*api.AccessRule, error) {
+							returnedPerm, err := tc.setFunc(path, permission, accountName)
+							perm = returnedPerm
+							return returnedPerm, err
+						},
+					},
+				}, tc.err
```

Comment:
We're returning the exact same error here as we're checking on line 238, making the check on line 238 always succeed.

If we split these two in: `clientErr` and `expectedErr`, this becomes immediately clear.

Reply:
Indeed, there was a problem with the check. Adjusted the tests properly.